waspc/data/Generator/templates/server/src/auth/providers/email/utils.ts

```diff
@@ -3,37 +3,30 @@ import { sign } from '../../../core/auth.js'
 import { emailSender } from '../../../email/index.js';
 import { Email } from '../../../email/core/types.js';
 import {
-  rethrowPossiblePrismaError,
+  rethrowPossibleAuthError,
   updateAuthIdentityProviderData,
   findAuthIdentity,
-  deserializeProviderData,
+  deserializeAndSanitizeProviderData,
 } from '../../utils.js';
 import waspServerConfig from '../../../config.js';
 import { type {= userEntityUpper =}, type {= authEntityUpper =} } from '../../../entities/index.js'
 
-type {= authEntityUpper =}Id = {= authEntityUpper =}['id']
-type {= userEntityUpper =}Id = {= userEntityUpper =}['id']
-
-type AuthWithId = {
-  id: {= authEntityUpper =}Id,
-}
-
-export async function createEmailVerificationLink(auth: AuthWithId, clientRoute: string) {
-  const token = await createEmailVerificationToken(auth);
+export async function createEmailVerificationLink(providerUserId: string, clientRoute: string) {
```

Comment:
Why don't we just say that it takes an email as first argument? If we know that is always going to be an email? Same goes for the rest of the functions.